File: app/src/main/java/stijnjj/hueapp/Activities/SettingsActivity.java
package stijnjj.hueapp.Activities;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import stijnjj.hueapp.R;
import stijnjj.hueapp.Settings;

public class SettingsActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        Settings s = Settings.createInstance(this);


        EditText username = (EditText) findViewById(R.id.txtUsername);
        username.setText(s.getUsername());

        EditText port = (EditText) findViewById(R.id.txtPort);
        port.setText(""+s.getPort());

        EditText location = (EditText) findViewById(R.id.txtLocation);
        location.setText(s.getLocation());

        Button connect = (Button) findViewById(R.id.btnConnect);
        connect.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent mActivity = new Intent(SettingsActivity.this, MainActivity.class);
                startActivity(mActivity);
            }
        });
    }
}

File: app/src/main/java/stijnjj/hueapp/Activities/SettingsActivity.kt
package stijnjj.hueapp.Activities

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText

import stijnjj.hueapp.R
import stijnjj.hueapp.Settings

class SettingsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        val s = Settings.createInstance(this)

        val username = findViewById<EditText>(R.id.txtUsername) as EditText
        username.setText(s.username)
        username.setOnFocusChangeListener { _, hasFocus ->
            if (!hasFocus)
                s.username = username.text.toString()
        }

        val port = findViewById<EditText>(R.id.txtPort) as EditText
        port.setText("" + s.port)
        port.setOnFocusChangeListener { _, hasFocus ->
            if (!hasFocus)
                s.port = port.text.toString().toInt()
        }

        val location = findViewById<EditText>(R.id.txtAddress) as EditText
        location.setText(s.location)
        location.setOnFocusChangeListener { _, hasFocus ->
            if (!hasFocus)
                s.location = location.text.toString()
        }

        val connect = findViewById<Button>(R.id.btnConnect) as Button
        connect.setOnClickListener {
            val mActivity = Intent(this@SettingsActivity, MainActivity::class.java)
            startActivity(mActivity)
        }
    }
}
